projects/cashmere/src/lib/accordion/accordion.component.spec.ts

```diff
@@ -0,0 +1,134 @@
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {AccordionComponent} from './accordion.component';
+import {AccordionToolbarComponent} from './accordion-toolbar.component';
+import {AccordionModule} from './accordion.module';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {Component} from '@angular/core';
+import {By} from '@angular/platform-browser';
+
+@Component({
+    template: `
+        <hc-accordion
+            [triggerAlign]="alignVal"
+            [hideToolbar]="hideVal"
+            [toolbarTrigger]="triggerVal"
+            [(open)]="openVal"
+            (openStart)="onOpenStart($event)"
+            (opened)="onOpen($event)"
+            (closeStart)="onCloseStart($event)"
+            (closed)="onClosed($event)"
+        >
+            <hc-accordion-toolbar>Toolbar</hc-accordion-toolbar>
+            Content
+        </hc-accordion>
+    `
+})
+export class TestAccordionComponent {
+    alignVal: string = 'left';
+    hideVal: boolean = false;
+    triggerVal: boolean = true;
+    openVal: boolean = false;
+
+    onOpenStart: (event?: Event) => void = () => {};
+    onOpen: (event?: Event) => void = () => {};
+    onCloseStart: (event?: Event) => void = () => {};
+    onClosed: (event?: Event) => void = () => {};
+}
+
+fdescribe('AccordionComponent', () => {
+    let component: TestAccordionComponent;
+    let fixture: ComponentFixture<TestAccordionComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [TestAccordionComponent],
+            imports: [AccordionModule, NoopAnimationsModule]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(TestAccordionComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should open the accordion when the open value is changed programtically', () => {
+        let accordionComponent = fixture.debugElement.query(By.directive(AccordionComponent));
+        expect(accordionComponent.nativeElement.classList.contains('hc-accordion-opened')).toBe(false);
+
+        component.openVal = true;
+        fixture.detectChanges();
+
+        expect(accordionComponent.nativeElement.classList.contains('hc-accordion-opened')).toBe(true);
+    });
+
+    it('should emit open events when the accordion open animation is complete', () => {
+        let accordionComponent = fixture.debugElement.query(By.directive(AccordionComponent));
+        spyOn(component, 'onOpen');
+        spyOn(component, 'onOpenStart');
+
+        accordionComponent.componentInstance.toggleOpen();
+        fixture.detectChanges();
+
+        fixture.whenStable().then(() => {
```

Comment:
Return at the start of this line

Reply:
Got it, so by adding return it ensures `whenStable` won't throw off the async of other tests?